x-pack/plugins/enterprise_search/public/applications/shared/unsaved_changes_prompt/unsaved_changes_prompt.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { Prompt } from 'react-router-dom';
+
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  hasUnsavedChanges: boolean;
+}
+
+export const UnsavedChangesPrompt: React.FC<Props> = ({ hasUnsavedChanges }) => {
+  const handler = (event: BeforeUnloadEvent) => {
+    if (hasUnsavedChanges) {
+      event.preventDefault();
+      event.returnValue = '';
```

Comment:
This doesn't make much sense to read, but is now you prompt a user whether or not they want to reload a page /  navigate away: https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload.

I don't think the message is customizable.

Reply:
Yeah, it's a bit confusing if you don't have a link to the doc. Should we add a comment in code?

Reply:
Done.